RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -1942,16 +1953,19 @@ class DeepTauId : public deep_tau::DeepTauBase {
           sp.scale(ele.deltaPhiSuperClusterTrackAtVtx(), dnn::ele_deltaPhiSuperClusterTrackAtVtx - e_index_offset);
       get(dnn::ele_deltaPhiSeedClusterTrackAtCalo + fill_index_offset_e) =
           sp.scale(ele.deltaPhiSeedClusterTrackAtCalo(), dnn::ele_deltaPhiSeedClusterTrackAtCalo - e_index_offset);
-      get(dnn::ele_mvaInput_earlyBrem + fill_index_offset_e) =
+      const bool mva_valid = (ele.mvaInput().earlyBrem > -2) || (year_ != 2026); // Known issue that input can be invalid in Phase2 samples (early/lateBrem==-2, hadEnergy==0, sigmaEtaEta/deltaEta==3.40282e+38). Unknown if also in Run2/3, so don't change there
```

Comment:
How this invalidity manifest itself, a NaN or an infinite number is returned? If so, a more generic way to check if the ele-mva is valid, e.g. using `std::isfinite`, will be useful. It will remove dependence on the specific requirement of `earlyBrem > -2` and a training version.

Reply:
In that case of being invalid, all of the 5 variables simply have the values I commented here. Otherwise (simply based on the values I've seen in 1-2 random samples from 2018), early/lateBrem is -1, 0 or 1, and the other 3 variables take small values (hadEnergy < 10000, eta variables < 10).

I would have thought that `3.40282e+38` classifies as a finite number, right? In that case, there are no NaNs or infinite numbers. Checking early/lateBrem to be not -2 seems like the safest check to do.

Reply:
Yes, `3.40282e+38` is finite, it equals to `std::numeric_limits<float>::max()`.
What about mva output for those inputs? Maybe it is testable, i.e. mva returns infinite, NaN or something similar? 
Anyway, check with `earlyBrem > -2` is probably fine, but in case one changes its meaningless default to something else it will cause difficult to debug issue in deepTau which I would like to avoid. I would also remove check of year as one can easily forget that it triggers the check of validity of electron mva inputs.

Reply:
I don't know about MVA outputs, but does it matter for this case? Since we're just giving the MVA inputs to DeepTau too. Or is there a way to get the MVA output value in this line?

The check of the year is because it was unknown for 2018v2p5 whether there were electrons with these bad inputs. From what I saw, it's not the case (i.e. inputs are always good), but in case there can be bad values, it was suggested to give the bad values to DeepTau regardless, because such events would also have "mistakenly" be added to the training. So, in the sense of: Better to keep it the way it is now for 2018v2p5, than to change it and worry about unforeseen consequences.

Reply:
> I don't know about MVA outputs, but does it matter for this case? Since we're just giving the MVA inputs to DeepTau too. Or is there a way to get the MVA output value in this line?

I mean that electron MVA output can be invalid when inputs are invalid and more straightforward for validity testing. It is just a guess.

> 
> The check of the year is because it was unknown for 2018v2p5 whether there were electrons with these bad inputs. From what I saw, it's not the case (i.e. inputs are always good), but in case there can be bad values, it was suggested to give the bad values to DeepTau regardless, because such events would also have "mistakenly" be added to the training. So, in the sense of: Better to keep it the way it is now for 2018v2p5, than to change it and worry about unforeseen consequences.

I understand this reasoning, but my worry is that one will perform raining with say 2030 conditions and the check will stop working. It is why I would like to have this check as general as possible. 
@kandrosov what do you think?

Reply:
There is no method that would indicate whatever `mvaInput` are valid in `GsfElectron`. The more robust way would be to change pat::Electron code adding such method and then use it here. We can use [default values](https://github.com/cms-sw/cmssw/blob/master/DataFormats/EgammaCandidates/interface/GsfElectron.h#L695-L701) set in the constructor to check the validity. E.g. 
```c++
struct MvaInput {
...
bool isValid() const
{
  static const MvaInput defaultMvaInput;
  return earlyBrem != defaultMvaInput.earlyBrem;
}
};
```

Reply:
Thanks @kandrosov, so let's stay with current implementation. We can ask in parallel EGamma people to implement such a validity flag.